src/par/src/autocluster.cpp

```diff
@@ -1256,27 +1256,24 @@ void AutoClusterMgr::printMacroCluster(Cluster* cluster_old, int& cluster_id)
 pair<float, float> AutoClusterMgr::printPinPos(Instance* macro_inst)
 {
   const float dbu = db_->getTech()->getDbUnitsPerMicron();
-  int lx = 100000000;
-  int ly = 100000000;
-  int ux = 0;
-  int uy = 0;
+  Rect bbox;
+  bbox.mergeInit();
   const Cell* cell = network_->cell(macro_inst);
   const char* cell_name = network_->name(cell);
   dbMaster* master = db_->findMaster(cell_name);
   for (dbMTerm* mterm : master->getMTerms()) {
-    if (mterm->getSigType() == 0) {
+    if (mterm->getSigType() == odb::dbSigType::SIGNAL) {
       for (dbMPin* mpin : mterm->getMPins()) {
         for (dbBox* box : mpin->getGeometry()) {
-          lx = min(lx, box->xMin());
-          ly = min(ly, box->yMin());
-          ux = max(ux, box->xMax());
-          uy = max(uy, box->yMax());
+          Rect rect;
+          box->getBox(rect);
+          bbox.merge(rect);
         }
       }
     }
   }
-  const float x_center = (lx + ux) / (2 * dbu);
-  const float y_center = (ly + uy) / (2 * dbu);
+  const float x_center = (bbox.xMin() + bbox.xMax()) / (2 * dbu);
+  const float y_center = (bbox.yMin() + bbox.yMax()) / (2 * dbu);
   return std::pair<float, float>(x_center, y_center);
```

Comment:
why use floats and not dbus?

Reply:
Because we are not using dbu in later macro placement stage.   It can be changed to dbu, but I will change many codes in different place.   I would suggest that we can fix this in the next generation Hier-RTL-MP.